sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/finishAnalysis.scala

```diff
@@ -38,6 +39,18 @@ object ReplaceExpressions extends Rule[LogicalPlan] {
   }
 }
 
+/**
```

Comment:
@cloud-fan We could also add these transformations in `ReplaceExpressions` rule and not require a dedicated rule (fyi).

Reply:
ah this sounds better!